src/components/Modals/PublishCollectionModal/PublishCollectionModal.tsx

```diff
@@ -1,39 +1,120 @@
 import * as React from 'react'
-import { ModalNavigation, Button } from 'decentraland-ui'
+import { ModalNavigation, Button, Mana, Loader } from 'decentraland-ui'
 import Modal from 'decentraland-dapps/dist/containers/Modal'
 import { t } from 'decentraland-dapps/dist/modules/translation/utils'
 
-import { Props } from './PublishCollectionModal.types'
+import { builder } from 'lib/api/builder'
+import { fromWei } from 'web3x-es/utils'
+import { RARITY_COLOR, ItemRarity } from 'modules/item/types'
+import { Props, State } from './PublishCollectionModal.types'
 import './PublishCollectionModal.css'
 
-export default class PublishCollectionModal extends React.PureComponent<Props> {
-  componentDidMount() {
+export default class PublishCollectionModal extends React.PureComponent<Props, State> {
+  state: State = { step: 1, rarities: [], isFetchingRarities: true }
+
+  async componentDidMount() {
     const { collection, onClose } = this.props
     if (!collection) {
       onClose()
     }
+
+    const rarities = await builder.fetchRarities()
+    this.setState({ rarities, isFetchingRarities: false })
   }
 
   handlePublish = () => {
     const { collection, items, onPublish } = this.props
     onPublish(collection!, items)
   }
 
+  handleProceed = () => {
+    this.setState({ step: 2 })
+  }
+
+  renderFirstStep = () => {
+    const { items } = this.props
+    const { rarities, isFetchingRarities } = this.state
+
+    const itemsByRarity: Record<string, { name: ItemRarity; count: number; price: number }> = {}
```

Comment:
I think this type should be indexed by ItemRarity, right? And, should we use rarity.id below?

Reply:
Same goes for this section: https://github.com/decentraland/builder/pull/1291/files#diff-9088b58a90f358c685624ad30bde921a26304e094bd9fe43ca4cef8007a82a33R69, reading the code the `name` seems less of a slug that what `id` does